src/frontend/index.js

```diff
@@ -28,24 +29,52 @@ logger.log(logPrefix + 'Loaded at ' + (new Date()));
 function init() {
 	var dispatcher = new EventEmitter();
 	
+	var config = null;
+	var dataHostnamesRoundRobin = 0;
```

Comment:
C++ backend does not return the list of hostnames, and the corresponding code is skipped. Only `backend_stub` returns the list and if you run `npm run start` it would fail without updating `/etc/hosts`.